File: src/main/java/com/team10/backend/domain/feed/entity/FeedLike.java
package com.team10.backend.domain.feed.entity;

import com.team10.backend.domain.user.entity.User;
import com.team10.backend.global.entity.BaseEntity;
import jakarta.persistence.*;

@Entity
@Table(name = "feed_likes", uniqueConstraints = {
        @UniqueConstraint(name = "uk_feed_like_post_user", columnNames = {"feed_post_id", "user_id"})
})
public class FeedLike extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "feed_post_id", nullable = false)
    private FeedPost feedPost;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    public FeedLike(FeedPost feedPost, User user) {
        this.feedPost = feedPost;
        this.user = user;
    }

    public FeedPost getFeedPost() {
        return this.feedPost;
    }

    public User getUser() {
        return this.user;
    }

    protected FeedLike() {
    }
}

File: src/main/java/com/team10/backend/domain/feed/entity/FeedLike.kt
package com.team10.backend.domain.feed.entity

import com.team10.backend.domain.user.entity.User
import com.team10.backend.global.entity.BaseEntity
import jakarta.persistence.*

@Entity
@Table(
    name = "feed_likes",
    uniqueConstraints = [UniqueConstraint(name = "uk_feed_like_post_user", columnNames = ["feed_post_id", "user_id"])]
)
class FeedLike(
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "feed_post_id", nullable = false)
    var feedPost: FeedPost,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    var user: User

) : BaseEntity()
